zetaclient/bitcoin/fee.go

```diff
@@ -0,0 +1,234 @@
+package bitcoin
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math"
+	"math/big"
+
+	"github.com/btcsuite/btcd/blockchain"
+	"github.com/btcsuite/btcd/btcjson"
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcutil"
+	"github.com/rs/zerolog"
+	"github.com/zeta-chain/zetacore/common"
+	"github.com/zeta-chain/zetacore/common/bitcoin"
+	clientcommon "github.com/zeta-chain/zetacore/zetaclient/common"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/pkg/errors"
+)
+
+const (
+	bytesPerKB              = 1000
+	bytesEmptyTx            = 10  // an empty tx is 10 bytes
+	bytesPerInput           = 41  // each input is 41 bytes
+	bytesPerOutputP2TR      = 43  // each P2TR output is 43 bytes
+	bytesPerOutputP2WSH     = 43  // each P2WSH output is 43 bytes
+	bytesPerOutputP2WPKH    = 31  // each P2WPKH output is 31 bytes
+	bytesPerOutputP2SH      = 32  // each P2SH output is 32 bytes
+	bytesPerOutputP2PKH     = 34  // each P2PKH output is 34 bytes
+	bytesPerOutputAvg       = 37  // average size of all above types of outputs (36.6 bytes)
+	bytes1stWitness         = 110 // the 1st witness incurs about 110 bytes and it may vary
+	bytesPerWitness         = 108 // each additional witness incurs about 108 bytes and it may vary
+	defaultDepositorFeeRate = 20  // 20 sat/byte is the default depositor fee rate
+
+	outTxBytesMin = uint64(239)  // 239vB == EstimateSegWitTxSize(2, 2, toP2WPKH)
+	outTxBytesMax = uint64(1543) // 1543v == EstimateSegWitTxSize(21, 2, toP2TR)
+	outTxBytesAvg = uint64(245)  // 245vB is a suggested gas limit for zeta core
+)
+
+var (
+	BtcOutTxBytesDepositor  uint64
+	BtcOutTxBytesWithdrawer uint64
+	DefaultDepositorFee     float64
+)
+
+func init() {
+	BtcOutTxBytesDepositor = OuttxSizeDepositor()   // 68vB, the outtx size incurred by the depositor
+	BtcOutTxBytesWithdrawer = OuttxSizeWithdrawer() // 177vB, the outtx size incurred by the withdrawer
+
+	// default depositor fee calculation is based on a fixed fee rate of 20 sat/byte just for simplicity.
+	// In reality, the fee rate on UTXO deposit is different from the fee rate when the UTXO is spent.
+	DefaultDepositorFee = DepositorFee(defaultDepositorFeeRate) // 0.00001360 (20 * 68vB / 100000000)
```

Comment:
Why not initializing with
```
var BtcOutTxBytesDepositor = OuttxSizeDepositor()
```

Reply:
https://github.com/zeta-chain/node/pull/1942/commits/2bcdfbc727c12612159ffbf008bb00505a996b4b